dataprofiler/profilers/column_profile_compilers.py

```diff
@@ -40,41 +41,90 @@ def _create_profile(self, df_series, options=None):
+            
             # Create profile if options allow for it or if there are no options
             if selected_col_profiles is None or \
-                    col_profile_type.col_type in selected_col_profiles:
-                col_profile_options = None
+               col_profile_type.col_type in selected_col_profiles:
+
+                col_options = None
                 if options and options.properties[col_profile_type.col_type]:
-                    col_profile_options = options.properties[col_profile_type.col_type]
-
-                try:
-                    self._profiles[col_profile_type.col_type] = \
-                        col_profile_type(df_series.name, options=col_profile_options)
-                    self._profiles[col_profile_type.col_type].update(df_series)
-                except Exception as e:
-                    import warnings
-                    warning_msg = "\n\n!!! WARNING Partial Profiler Failure !!!\n\n"
-                    warning_msg += "Profiling Type: {}".format(col_profile_type.col_type)
-                    warning_msg += "\nException: {}".format(type(e).__name__)
-                    warning_msg += "\nMessage: {}".format(e)
-
-                    # This is considered a major error
-                    if type(e).__name__ == "ValueError":
-                        raise ValueError(e)
+                    col_options = options.properties[col_profile_type.col_type]
                     
-                    warning_msg += "\n\nFor labeler errors, try installing "
-                    warning_msg += "the extra ml requirements via:\n\n"
-                    warning_msg += "$ pip install dataprofiler[ml] --user\n\n"
-
-                    warnings.warn(warning_msg, RuntimeWarning, stacklevel=2)
+                self._profiles[col_profile_type.col_type] = \
+                    col_profile_type(df_series.name, options=col_options)
+                
+                # MULTIPROCESSING
+                single_process_flag = True
+                if multiprocess_flag:
+                    single_process_flag = False
+                    try:                        
+                        f = pool.apply_async(
+                            self._profiles[col_profile_type.col_type].update,
+                            (df_series,)
+                        )
+                        multi_process_dict[col_profile_type.col_type] = f
+                    except Exception as e:
+                        # Occurs when object cannot be pickled
+                        single_process_flag = True
+
+                # Add to single process list to execute later
+                if single_process_flag:
+                    single_process_list.append(col_profile_type.col_type)
+
+        for col in multi_process_dict.keys():
+            try:
+                f = multi_process_dict[col]
+                returned_profile = f.get()
+                if returned_profile is not None:
```

Comment:
Why have this outside of the other for loop? Is it to wait for the process to finish and validate after the thread has finished? If so, shouldnt this be after  pool.close() and pool.join()?